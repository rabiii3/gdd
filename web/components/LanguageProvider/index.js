import { compose } from 'recompose';
import { connect } from 'react-redux';
import component from './component';
import { getLocale } from '../../selectors/language';

const mapStateToProps = state => ({
  locale: getLocale(state),
});

export const enhance = compose(connect(mapStateToProps));

export default enhance(component);
