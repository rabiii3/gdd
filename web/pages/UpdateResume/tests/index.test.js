import React from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import LanguageProvider from '../../../components/LanguageProvider';
import configureStore from '../../../store/configureStore.test';
import { enhance } from '../';

describe('web |Â pages | UpdateResume | index', () => {
  describe('render', () => {
    it('should render', () => {
      const Component = () => <div />;
      const EnhancedComponent = enhance(Component);
      const state = {
        auth:{
          user:{},
          token:"token",
        },
        config:{
          tenant:{
            key: "key",
          }
        },
        people:{},
        resumes:{},
        comments:{},
        language: { locale: 'en' }
      };
      const store = configureStore(state);
      const history = createBrowserHistory();
      const match = { params: { id: 1 } };

      const wrapper = mount(
        <Provider store={store}>
          <LanguageProvider messages={{}} >
            <Router history={history}>
              <EnhancedComponent match={match} />
            </Router>
          </LanguageProvider>
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
