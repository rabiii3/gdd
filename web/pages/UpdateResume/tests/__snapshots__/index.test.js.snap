// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web |Â pages | UpdateResume | index render should render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(LanguageProvider)
    messages={Object {}}
  >
    <LanguageProvider
      dispatch={[Function]}
      locale="en"
      messages={Object {}}
    >
      <IntlProvider
        key="en"
        locale="en"
      >
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <WithStyles(withStateHandlers(withStateHandlers(withStateHandlers(Connect(lifecycle(EnhancedComponent))))))
            match={
              Object {
                "params": Object {
                  "id": 1,
                },
              }
            }
          >
            <withStateHandlers(withStateHandlers(withStateHandlers(Connect(lifecycle(EnhancedComponent)))))
              classes={
                Object {
                  "bold": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------bold-5",
                  "comment": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------comment-8",
                  "flexGrow": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------flexGrow-3",
                  "grey": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------grey-4",
                  "heading": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------heading-7",
                  "iconHeader": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------iconHeader-6",
                  "left": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------left-9",
                  "margin": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------margin-2",
                  "p": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------p-11",
                  "paper": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------paper-1",
                  "user": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------user-10",
                }
              }
              match={
                Object {
                  "params": Object {
                    "id": 1,
                  },
                }
              }
            >
              <withStateHandlers(withStateHandlers(Connect(lifecycle(EnhancedComponent))))
                classes={
                  Object {
                    "bold": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------bold-5",
                    "comment": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------comment-8",
                    "flexGrow": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------flexGrow-3",
                    "grey": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------grey-4",
                    "heading": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------heading-7",
                    "iconHeader": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------iconHeader-6",
                    "left": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------left-9",
                    "margin": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------margin-2",
                    "p": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------p-11",
                    "paper": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------paper-1",
                    "user": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------user-10",
                  }
                }
                isEditable={false}
                match={
                  Object {
                    "params": Object {
                      "id": 1,
                    },
                  }
                }
                setEditable={[Function]}
              >
                <withStateHandlers(Connect(lifecycle(EnhancedComponent)))
                  classes={
                    Object {
                      "bold": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------bold-5",
                      "comment": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------comment-8",
                      "flexGrow": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------flexGrow-3",
                      "grey": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------grey-4",
                      "heading": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------heading-7",
                      "iconHeader": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------iconHeader-6",
                      "left": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------left-9",
                      "margin": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------margin-2",
                      "p": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------p-11",
                      "paper": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------paper-1",
                      "user": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------user-10",
                    }
                  }
                  isDeleted={false}
                  isEditable={false}
                  match={
                    Object {
                      "params": Object {
                        "id": 1,
                      },
                    }
                  }
                  setDeleting={[Function]}
                  setEditable={[Function]}
                >
                  <Connect(lifecycle(EnhancedComponent))
                    classes={
                      Object {
                        "bold": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------bold-5",
                        "comment": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------comment-8",
                        "flexGrow": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------flexGrow-3",
                        "grey": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------grey-4",
                        "heading": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------heading-7",
                        "iconHeader": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------iconHeader-6",
                        "left": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------left-9",
                        "margin": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------margin-2",
                        "p": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------p-11",
                        "paper": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------paper-1",
                        "user": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------user-10",
                      }
                    }
                    expanded={false}
                    handleExpandClick={[Function]}
                    isDeleted={false}
                    isEditable={false}
                    match={
                      Object {
                        "params": Object {
                          "id": 1,
                        },
                      }
                    }
                    setDeleting={[Function]}
                    setEditable={[Function]}
                  >
                    <lifecycle(EnhancedComponent)
                      actions={
                        Object {
                          "addComment": [Function],
                          "deleteComment": [Function],
                          "deleteResume": [Function],
                          "loadFile": [Function],
                          "updateComment": [Function],
                          "updateResume": [Function],
                        }
                      }
                      classes={
                        Object {
                          "bold": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------bold-5",
                          "comment": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------comment-8",
                          "flexGrow": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------flexGrow-3",
                          "grey": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------grey-4",
                          "heading": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------heading-7",
                          "iconHeader": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------iconHeader-6",
                          "left": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------left-9",
                          "margin": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------margin-2",
                          "p": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------p-11",
                          "paper": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------paper-1",
                          "user": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------user-10",
                        }
                      }
                      comments={Array []}
                      expanded={false}
                      handleExpandClick={[Function]}
                      isDeleted={false}
                      isEditable={false}
                      match={
                        Object {
                          "params": Object {
                            "id": 1,
                          },
                        }
                      }
                      people={Object {}}
                      resumes={Array []}
                      setDeleting={[Function]}
                      setEditable={[Function]}
                      tenant="key"
                      token="token"
                    >
                      <EnhancedComponent
                        actions={
                          Object {
                            "addComment": [Function],
                            "deleteComment": [Function],
                            "deleteResume": [Function],
                            "loadFile": [Function],
                            "updateComment": [Function],
                            "updateResume": [Function],
                          }
                        }
                        classes={
                          Object {
                            "bold": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------bold-5",
                            "comment": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------comment-8",
                            "flexGrow": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------flexGrow-3",
                            "grey": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------grey-4",
                            "heading": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------heading-7",
                            "iconHeader": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------iconHeader-6",
                            "left": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------left-9",
                            "margin": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------margin-2",
                            "p": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------p-11",
                            "paper": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------paper-1",
                            "user": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------user-10",
                          }
                        }
                        comments={Array []}
                        expanded={false}
                        handleExpandClick={[Function]}
                        isDeleted={false}
                        isEditable={false}
                        match={
                          Object {
                            "params": Object {
                              "id": 1,
                            },
                          }
                        }
                        people={Object {}}
                        resumes={Array []}
                        setDeleting={[Function]}
                        setEditable={[Function]}
                        tenant="key"
                        token="token"
                      >
                        <Component
                          actions={
                            Object {
                              "addComment": [Function],
                              "deleteComment": [Function],
                              "deleteResume": [Function],
                              "loadFile": [Function],
                              "updateComment": [Function],
                              "updateResume": [Function],
                            }
                          }
                          classes={
                            Object {
                              "bold": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------bold-5",
                              "comment": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------comment-8",
                              "flexGrow": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------flexGrow-3",
                              "grey": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------grey-4",
                              "heading": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------heading-7",
                              "iconHeader": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------iconHeader-6",
                              "left": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------left-9",
                              "margin": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------margin-2",
                              "p": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------p-11",
                              "paper": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------paper-1",
                              "user": "withStateHandlers-withStateHandlers-withStateHandlers-Connect-lifecycle-EnhancedComponent------user-10",
                            }
                          }
                          comments={Array []}
                          expanded={false}
                          handleExpandClick={[Function]}
                          isDeleted={false}
                          isEditable={false}
                          match={
                            Object {
                              "params": Object {
                                "id": 1,
                              },
                            }
                          }
                          people={Object {}}
                          resumes={Array []}
                          setDeleting={[Function]}
                          setEditable={[Function]}
                          tenant="key"
                          token="token"
                        >
                          <div />
                        </Component>
                      </EnhancedComponent>
                    </lifecycle(EnhancedComponent)>
                  </Connect(lifecycle(EnhancedComponent))>
                </withStateHandlers(Connect(lifecycle(EnhancedComponent)))>
              </withStateHandlers(withStateHandlers(Connect(lifecycle(EnhancedComponent))))>
            </withStateHandlers(withStateHandlers(withStateHandlers(Connect(lifecycle(EnhancedComponent)))))>
          </WithStyles(withStateHandlers(withStateHandlers(withStateHandlers(Connect(lifecycle(EnhancedComponent))))))>
        </Router>
      </IntlProvider>
    </LanguageProvider>
  </Connect(LanguageProvider)>
</Provider>
`;
